jetty-webapp/src/main/java/org/eclipse/jetty/webapp/CachingWebAppClassLoader.java

```diff
@@ -115,12 +152,65 @@ public URL getResource(String name)
     public void clearCache()
     {
         _cache.clear();
+        _classCache.clear();
         _notFound.clear();
     }
 
+    public void setUseCache(boolean useCache)
+    {
+            this.useCache = useCache;
+    }
+
     @Override
     public String toString()
     {
         return "Caching[" + super.toString() + "]";
     }
+
+    /**
+     * Lifecycle listener, that can be used from {@link WebAppContext#addLifeCycleListener(LifeCycle.Listener)}
+     * to switch off the {@link #_cache} and {@link #_classCache} after web application has been started.
```

Comment:
This seems like a feature that should be exposed via JMX, not as a LifeCycleListener.

Reply:
Ok, it can be used from JMX, but in case of embedded jetty developer can also use this listener from the WebAppContext instance via `WebAppContext#addLifeCycleListener.` So I suggest to update javadoc for both possibilities. In my app, I add the listener to the WebAppContext via java code.